packages/editor/src/components/post-schedule/label.js

```diff
@@ -5,15 +5,31 @@ import { __ } from '@wordpress/i18n';
 import { dateI18n, getSettings } from '@wordpress/date';
 import { withSelect } from '@wordpress/data';
 
-function PostScheduleLabel( { date } ) {
+// Unlike in the PHP backend, in the REST API response draft posts for which a
+// date has not yet been set return a full gmt date string instead of the empty
+// 0000-00-00 00:00:00 placeholder used in the database. To infer that a post
+// is set to publish "immediately" we check whether the date and modified date
+// are matching values.
+function isFloatingDate( { date, modified, status } ) {
+	if ( status === 'draft' || status === 'auto-draft' ) {
+		return date === modified;
+	}
+	return false;
+}
+
+export function PostScheduleLabel( props ) {
+	const { date } = props;
+	const floating = isFloatingDate( props );
 	const settings = getSettings();
-	return date ?
+	return date && ! floating ?
 		dateI18n( settings.formats.datetime, date ) :
 		__( 'Immediately' );
 }
 
 export default withSelect( ( select ) => {
 	return {
 		date: select( 'core/editor' ).getEditedPostAttribute( 'date' ),
```

Comment:
What if instead of having to do this check in the component, we do it in the selector `getEditedPostAttribute`  instead returning `null` or `undefined` for the `date` attribute if the condition above is met? I'm wondering if a more global change like that is better as it save us having to repeat the same operation if we have to check again in a separate component if needed.

Alternatively, if we don't want to alter the "date" property value "virtually", should we add a new `editedPostHasFloatingDate` selector or something like that?

Reply:
That makes a lot of sense to me. Us having to deal with this in the component seems odd (and probably what introduced this regression, right?)

Reply:
I'm actually leaning towards the second option: the separate `editedPostHasFloatingDate` selector because if normalize the "date" property, it should probably be better normalized before it hits the reducer which might be more complex to achieve consistently (several ways to load posts)

Reply:
I think I'm in favor of that option as well, because although in a new WP ticket we do intend to support passing `null` to put a post _back_ into floating date status, given the way the API response is constructed a post object will continue to return a date string in all circumstances, and not handling an alternative virtual type in the store feels cleaner.